test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts

```diff
@@ -92,11 +101,55 @@ contract('DCAStrategiesManagementHandler', () => {
         expect(strategyId).to.be.equal(1);
         expect(strategyName).to.be.equal(strategy.name);
         expect(owner).to.be.equal(strategy.owner);
-        expect(SHARES.length).to.be.equal(strategy.tokens.length);
-        SHARES.forEach((s: any, i: any) => {
-          expect(tokens[i].share).to.be.equal(s.share);
-          expect(tokens[i].token).to.be.equal(s.token);
-        });
+        compareTokens(SHARES, tokens);
+      });
+    });
+  });
+
+  describe('createStrategy', () => {
+    let owner: Wallet;
+    const NAME = ethers.utils.formatBytes32String('Optimism Ecosystem - v1');
+
+    const SHARE_TOKEN_A_1 = { token: generateRandomAddress(), share: BigNumber.from(50) };
+    const SHARE_TOKEN_B_1 = { token: generateRandomAddress(), share: BigNumber.from(50) };
+    const SHARES_1 = [SHARE_TOKEN_A_1, SHARE_TOKEN_B_1];
+
+    const SHARE_TOKEN_A_2 = { token: constants.NOT_ZERO_ADDRESS, share: BigNumber.from(30) };
+    const SHARE_TOKEN_B_2 = { token: constants.NOT_ZERO_ADDRESS, share: BigNumber.from(40) };
+    const SHARE_TOKEN_C_2 = { token: generateRandomAddress(), share: BigNumber.from(30) };
+    const SHARES_2 = [SHARE_TOKEN_A_2, SHARE_TOKEN_B_2, SHARE_TOKEN_C_2];
+
+    given(async () => {
+      owner = await wallet.generateRandom();
+    });
+    when('sender is not the owner', () => {
+      given(async () => {
+        await DCAStrategiesManagementHandlerMock.createStrategy(NAME, SHARES_1, owner.address);
+      });
+      then('tx reverted with message', async () => {
+        await expect(DCAStrategiesManagementHandlerMock.updateStrategyTokens(1, SHARES_2)).to.be.revertedWith('OnlyStratOwner()');
+      });
+    });
+    when('strategy is updated', () => {
```

Comment:
Could you also test that the owner and name have not changed? Just to be on the safe side